Exploration Queries/DnsEvents/LeastPrevClientIP-DNSNameQueryToIP.txt

```diff
@@ -1,27 +1,23 @@
-// Name: DNS Name Lookup Query from least prevalent ClientIP to remote IP Address
-// Description: Summary of Bottom 10 Client IP and Domain Names for a given remote IPAddress from DnsEvent Lookup Query data.
-//
-// Entity: IPAddress
-// Input: IPAddress
-// Output: ClientIP
-//
-// QueryPeriod:  +-3h
-//
-// Data Source: #DnsEvents
-//
+// Id: 801bacb0-612a-4195-a84f-7939cca63b92
+// DisplayName: DNS Name Lookup Query from least prevalent ClientIP to remote IP Address
+// Description: Summary of Bottom 10 Client IP and Domain Names for a given remote IPAddress from DnsEvent Lookup Query data (set time range to +-3h when running the query)
+// InputEntityType: Ip
+// InputFields: [Address]
+// OutputEntityTypes: [Ip]
+// QueryPeriodBefore: 3h
+// QueryPeriodAfter: 3h
+// DataSources: #DnsEvents
 // Tactics: #CommandAndControl, #Exfiltration
-//
-let GetAllIPByClientIP = (suspiciousEventTime:datetime, v_IPAddress:string){
-let v_StartTime = suspiciousEventTime-3h;
-let v_EndTime = suspiciousEventTime+3h;
+
+let GetAllIPByClientIP = (v_IP_Address:string){
 DnsEvents
-| where TimeGenerated between (v_StartTime .. v_EndTime)
-| where SubType == "LookupQuery"
-| where IPAddresses contains v_IPAddress
+| where SubType == 'LookupQuery'
+| where IPAddresses contains v_IP_Address
 | summarize min(TimeGenerated), max(TimeGenerated), makeset(Name), count() by ClientIP, IPAddresses
-| project StartTimeUtc = min_TimeGenerated, EndTimeUtc = max_TimeGenerated, ClientIP, DomainNames = set_Name, IPAddresses, count_ 
+| project StartTimeUtc = min_TimeGenerated, EndTimeUtc = max_TimeGenerated, ClientIP, count_
 | top 10 by count_ asc nulls last
-| project StartTimeUtc, EndTimeUtc, ClientIP, IPAddresses, DomainNames
+| project StartTimeUtc, EndTimeUtc, ClientIP
+| project-rename IP_Address=ClientIP
```

Comment:
rename the relevant columns to the format {V3EntityType_FieldName}

Reply:
This query now results in useless info.  It simply returning the IP you looked up.  The goal here is to look up an IP and see what Domain Names are associated with it.  Needs to include -

| project StartTimeUtc = min_TimeGenerated, EndTimeUtc = max_TimeGenerated, ClientIP, DomainNames = set_Name, count_
| top 10 by count_ asc nulls last
| project StartTimeUtc, EndTimeUtc, ClientIP, DomainNames

Reply:
I should say, it is returning the ClientIP's who did the lookup, but not the associated Domains.  This is very relevant and should be included in the result set

Reply:
Clarifying a bit here, I think it would be good if we could pack the domainname with the IPaddress and then be able to pivot on both as it is possible that domainname has multiple IP's

Reply:
True, this is an issue that we were trying to solve recently. Please have a look at the mail that I sent you. 

I want to be sure that I did not affect the query incorrectly, so I would like you to have a look and check that everything is good. 

Once we fully agree on the solution I will add it to all the queries